rest/config.go

```diff
@@ -814,21 +817,29 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 
 // expandEnv replaces $var or ${var} in config according to the values of the
 // current environment variables. The replacement is case-sensitive. References
-// to undefined variables are replaced by an empty string. A default value can
+// to undefined variables will result in an error. A default value can
 // be given by using the form ${var:-default value}.
-func expandEnv(config []byte) []byte {
+func expandEnv(config []byte) (value []byte, errs error) {
 	return []byte(os.Expand(string(config), func(key string) string {
 		if key == "$" {
 			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
 			return key
 		}
-		return envDefaultExpansion(key, os.Getenv)
-	}))
+		val, err := envDefaultExpansion(key, os.Getenv)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+		return val
+	})), errs
 }
 
+// ErrEnvVarUndefined is returned when a specified variable can’t be resolved from the system environment or no
+// default value is supplied in the configuration — startup will be aborted.
```

Comment:
SG startup behaviour isn't really related to this error.

```suggestion
// ErrEnvVarUndefined is returned when a specified variable can’t be resolved from the system environment and no
// default value is supplied in the configuration.
```

Reply:
Actually the intention was to include a note saying that SG won't allow even a single undefined variable being referenced in the config file without a default value. This comment has been refactored.

Reply:
I think I was more concerned with commenting the current behaviour in a place where we're not actually aborting startup (i.e. returning the error).

If we switched SG to log a warning instead this comment would likely be left out of date because it's not really part of the code.